File: ios/Zest/Services/StorageService.swift
import Foundation
import FirebaseStorage

class StorageService {
    static let shared = StorageService()
    private let storage = Storage.storage()
    
    private init() {}
    
    func uploadRecipeImage(imageData: Data, recipeId: String) async throws -> String {
        let timestamp = Int(Date().timeIntervalSince1970)
        let fileName = "recipe_\(timestamp).jpg"
        let storageRef = storage.reference().child("recipes/\(recipeId)/\(fileName)")
        
        _ = try await storageRef.putDataAsync(imageData)
        let downloadURL = try await storageRef.downloadURL()
        return downloadURL.absoluteString
    }
    
    func uploadUserProfileImage(imageData: Data, userId: String) async throws -> String {
        let timestamp = Int(Date().timeIntervalSince1970)
        let fileName = "profile_\(timestamp).jpg"
        let storageRef = storage.reference().child("users/\(userId)/profile/\(fileName)")
        
        _ = try await storageRef.putDataAsync(imageData)
        let downloadURL = try await storageRef.downloadURL()
        return downloadURL.absoluteString
    }
    
    func uploadCookbookCover(imageData: Data, cookbookId: String) async throws -> String {
        let timestamp = Int(Date().timeIntervalSince1970)
        let fileName = "cover_\(timestamp).jpg"
        let storageRef = storage.reference().child("cookbooks/\(cookbookId)/\(fileName)")
        
        _ = try await storageRef.putDataAsync(imageData)
        let downloadURL = try await storageRef.downloadURL()
        return downloadURL.absoluteString
    }
    
    func deleteImage(imageUrl: String) async throws {
        let storageRef = storage.reference(forURL: imageUrl)
        try await storageRef.delete()
    }
}

File: android/app/src/main/java/com/neostudios/zest/data/remote/StorageService.kt
package com.neostudios.zest.data.remote

import android.net.Uri
import com.google.firebase.storage.FirebaseStorage
import kotlinx.coroutines.tasks.await
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class StorageService @Inject constructor(
    private val storage: FirebaseStorage
) {
    
    suspend fun uploadRecipeImage(uri: Uri, recipeId: String): String {
        val timestamp = System.currentTimeMillis()
        val fileName = "recipe_${timestamp}.jpg"
        val storageRef = storage.reference.child("recipes/$recipeId/$fileName")
        
        storageRef.putFile(uri).await()
        return storageRef.downloadUrl.await().toString()
    }
    
    suspend fun uploadUserProfileImage(uri: Uri, userId: String): String {
        val timestamp = System.currentTimeMillis()
        val fileName = "profile_${timestamp}.jpg"
        val storageRef = storage.reference.child("users/$userId/profile/$fileName")
        
        storageRef.putFile(uri).await()
        return storageRef.downloadUrl.await().toString()
    }
    
    suspend fun uploadCookbookCover(uri: Uri, cookbookId: String): String {
        val timestamp = System.currentTimeMillis()
        val fileName = "cover_${timestamp}.jpg"
        val storageRef = storage.reference.child("cookbooks/$cookbookId/$fileName")
        
        storageRef.putFile(uri).await()
        return storageRef.downloadUrl.await().toString()
    }
    
    suspend fun deleteImage(imageUrl: String) {
        try {
            val storageRef = storage.getReferenceFromUrl(imageUrl)
            storageRef.delete().await()
        } catch (e: Exception) {
            // Handle error silently
        }
    }
}
